backend/clickhouse/log_row.go

```diff
@@ -0,0 +1,104 @@
+package clickhouse
+
+import (
+	"strconv"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/highlight/highlight/sdk/highlight-go"
+	"github.com/influxdata/influxdb-client-go/v2/log"
+)
+
+type LogRowPrimaryAttrs struct {
+	Timestamp       time.Time
+	ProjectId       uint32
+	TraceId         string
+	SpanId          string
+	SecureSessionId string
+}
+
+type LogRow struct {
+	LogRowPrimaryAttrs
+	UUID           string
+	TraceFlags     uint32
+	SeverityText   string
+	SeverityNumber int32
+	ServiceName    string
+	Body           string
+	LogAttributes  map[string]string
+}
+
+func NewLogRow(attrs LogRowPrimaryAttrs, opts ...LogRowOption) *LogRow {
+	logRow := &LogRow{
+		LogRowPrimaryAttrs: LogRowPrimaryAttrs{
+			Timestamp:       attrs.Timestamp,
+			TraceId:         attrs.TraceId,
+			SpanId:          attrs.SpanId,
+			ProjectId:       attrs.ProjectId,
+			SecureSessionId: attrs.SecureSessionId,
+		},
+		UUID:           uuid.New().String(),
+		SeverityText:   "INFO",
+		SeverityNumber: int32(log.InfoLevel),
+	}
+
+	for _, opt := range opts {
+		opt(logRow)
+	}
+
+	return logRow
+}
+
+func (l *LogRow) Cursor() string {
+	return encodeCursor(l.Timestamp, l.UUID)
+}
+
+type LogRowOption func(*LogRow)
+
+func WithLogAttributes(resourceAttributes, eventAttributes map[string]any) LogRowOption {
+	return func(h *LogRow) {
+		h.LogAttributes = getAttributesMap(resourceAttributes, eventAttributes)
```

Comment:
Ideally, we can make things like `LogAttributes` private (`logAttributes`) and external code can only use these `With*` functions to set the value.

For example, this would help with something like #4464 where we could write code like:

```go
logRow1 := NewLogRow(
  WithSeverityText("INFO")
)
logRow2 := NewLogRow(
  WithSeverityText("info")
)
```

and the underlying code in `WithSeverityText` would handle normalizing to some agreed format.